Deployment II

reactivepixel opened the pull request.

THEJNIGH committed:
db file stuff

THEJNIGH committed:
hello?

THEJNIGH committed:
Database stuff

reactivepixel committed:
Add Unit Test Example

therealaldo committed:
Added License & Contribute files

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity into order_bot

therealaldo committed:
listed team members

therealaldo committed:
fixed request and response objects in readme

therealaldo committed:
fixed readme

THEJNIGH committed:
added model folder

reactivepixel committed:
Add Unit Tests

THEJNIGH committed:
Setup sequelize connection.

THEJNIGH committed:
Added database installation to readme.

Commit:
Created models and tables for database

Commit:
changed readme

therealaldo committed:
updated database
added database documentation
the table schemas could be synced a better way

therealaldo committed:
trying to fix onflicts

therealaldo committed:
fixed readme

reactivepixel commented on CONTRIBUTE.md:
```diff
@@ -0,0 +1,99 @@
+This sections explains how bug triaging is done for your project. Help beginners by including examples to good bug reports and providing them questions they should look to answer.
+
+* You can help report bugs by filing them here:
+* You can look through the existing bugs here:
```
Where can bugs be reported? And where are they tracked?

reactivepixel commented on package.json:
```diff
@@ -19,9 +20,15 @@
   "homepage": "https://github.com/reactivepixel/Gravity#readme",
   "dependencies": {
     "body-parser": "1.15.0",
+    "dotenv": "2.0.0",
```
This should be a dev only dependency

reactivepixel commented on routes/order.js:
```diff
@@ -19,7 +19,7 @@ module.exports = function (express) {
       data.timestamp = timestamp.makeTimestamp();
       // ending the response and console logging the response data
-      console.log(data);
+      res.send({healthy: true});
```
This is just a demo data for showing how Unit Testing is handled. This should be switched back to actually displaying the data generated upon request

therealaldo replied:
With how the routes are set up, how do you display the new data to the user? Is it maybe because we are using a get method before using the put method?

reactivepixel commented on server/db.js:
```diff
@@ -0,0 +1,18 @@
+module.exports = function() {
+
+  //Requring dependencies.
+  const dotenv = require('dotenv').load();
```
If the .env file is not present (on staging, QA, and production servers) Will this Fatally Error?

therealaldo replied:
@jacobnigh @trichh Can you guys check this out?

THEJNIGH replied:
Yes, it will error out, but my next question is how to use the 'process.env.DB_WHATEVER' without including the .env file. Brad said it would be with server config. So would we have to give the server the variables so that we would no longer have to use the .env file? @reactivepixel

therealaldo committed:
added use strict to all necessary files

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

therealaldo committed:
remove

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

therealaldo committed:
added dotenv to devDependencies

therealaldo committed:
added dotenv to dev dependencies

thisisbrad commented on package.json:
```diff
@@ -21,7 +22,13 @@
     "body-parser": "1.15.0",
     "express": "4.13.4",
     "faker": "3.0.1",
```
Should faker.js be a development dependency as well?

therealaldo committed:
added faker to dev dependencies

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

therealaldo committed:
Added procfile for starting heroku

Commit:
moved create table functions to db.js so you only have to run 1 file to create all the tables

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

Commit:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity into order_bot

therealaldo committed:
changed package.json

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

therealaldo committed:
Cleaning up code
fixing typos and inconsistencies
changed itemsOrdered to orderedItems
edited comments to be more descriptive

therealaldo committed:
pulled from order_bot

reactivepixel commented on models/orderedItems.js:
```diff
@@ -0,0 +1,9 @@
+'use strict';
```
File names should all be lower case

reactivepixel commented on models/orderedItems.js:
```diff
@@ -0,0 +1,9 @@
+'use strict';
+module.exports = function() {
+	const db = require('../server/db.js')();
+  const Sequelize = require('sequelize');
```
Spacing issue (no tabs vs tabs)

reactivepixel commented on models/units.js:
```diff
@@ -0,0 +1,9 @@
+  const sequelize = db.connection;
+
+  // Defining units based on the table schema from the db.js
+	const units = db.units;
```
Not constant. Check others similar to this structure.

reactivepixel commented on server/db.js:
```diff
@@ -0,0 +1,95 @@
+
+  // Initializing sequelize
+  const sequelize = new Sequelize(process.env.DB_NAME, process.env.DB_USER, process.env.DB_PASS, {
+    host: 'localhost',
```
Change to Env Var

reactivepixel commented on server/server.js:
```diff
@@ -3,10 +3,14 @@
+<<<<<<< HEAD
+=======
+// Adding body parser to parse json automatically
+>>>>>>> 3331767e9feb3862b3c14d53c8eb96caf918ca24
```
Unresolved Merge Conflict

therealaldo committed:
fixed small errors

therealaldo committed:
fixing errors from pull request

therealaldo committed:
Merge branch 'order_bot' of https://github.com/reactivepixel/Gravity

therealaldo committed:
fixed errors from pull request

Keownc committed:
Created an unit test to test if the uuid and the timestamp to  is matching the json data

Commit:
Getting rid of errors in models

Commit:
inserts items into database

therealaldo committed:
first attempt at order/create

therealaldo committed:
creating order and orderedItem
this makes sense at first glance but it hasn't been tested so this could break completely and not even be the right way to do this

therealaldo committed:
changed request hits to 1

Commit:
Adding data to the database dynamically

Commit:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

therealaldo committed:
created unit test file

therealaldo committed:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

THEJNIGH committed:
started unit testing db

THEJNIGH committed:
add chai

THEJNIGH committed:
Unit testing yay

Keownc committed:
Combine the new test into the default test

therealaldo committed:
committing changes

therealaldo committed:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

therealaldo committed:
deleted file

therealaldo committed:
changed recipients to recipient

jamesalexanderdickerson committed:
change test environment port to 1234 so that you can run the server and the unit test at the same time.

therealaldo committed:
merging unittest

therealaldo committed:
trying to push unittest

THEJNIGH committed:
OH NO, SOMETHING BROKE

THEJNIGH committed:
Merge conflict:

therealaldo committed:
Corrected Models and PUT Route
Added crud functionality to table models
modified put route to give success and error messages
better handling of the data transfer

therealaldo committed:
fixed syntax

THEJNIGH committed:
working on new routes

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

therealaldo committed:
order route is working with new models
route could be optimized using transactions/migrations/foreign keys

therealaldo committed:
removed unnecessary let variable

THEJNIGH committed:
Create unit testing works

therealaldo committed:
changed route and added foreign key

therealaldo committed:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

THEJNIGH committed:
SOMETHING BROKE WTF

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

jamesalexanderdickerson committed:
changed model for orders, orderedItems, and db.js

therealaldo committed:
fixed models, route
need to fix route where you can access the saved data from the database to show it back to the user

therealaldo committed:
removed commetned route

chrisfortin16 committed:
changed inventory model and units model

chrisfortin16 committed:
fixed postinstall on package.json file

THEJNIGH committed:
My test works?

therealaldo committed:
changed route

therealaldo committed:
fixed merge conflicts in test

therealaldo committed:
files from google hangout

therealaldo committed:
trying to fix route

THEJNIGH committed:
Added findall to orders route

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

chrisfortin16 committed:
Fixed models to find where data = payload

THEJNIGH committed:
Unit tested all of order route CRUD

THEJNIGH committed:
Unit tested routes
CRUD is all tested.

THEJNIGH committed:
Fixed merge conflict
Fixed in orders.js

therealaldo committed:
added async to packages
also commented out the put request to work on enhanced version

therealaldo committed:
Completed PUT route using async.waterfall
Added async.waterfall method to handle the creation of orders and orderedItems
Used .waterfall() to handle the final response with the newly created data from the db
added comments explaining thought process and methods used along with reasoning for using them

therealaldo committed:
Enhanced comments
added documentation link for async module

Commit:
Added inventory route

Commit:
fixed inventory model and added inventory route in server.js

THEJNIGH committed:
added testing file for ordered items

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

Commit:
fixed syntax error on inventory route

Commit:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

Commit:
added units route and fixed units model

THEJNIGH committed:
Working on orderedItems route and unit testing

THEJNIGH committed:
Fixed merge conflict

therealaldo committed:
Modified the variables which were being saved
SavedData was created in the inventory and units put request but was being left behind since it wasn't in one of the callbacks

therealaldo committed:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

therealaldo committed:
fixed syntax errors
function was misspelled a couple times and some semi colons were missing

therealaldo committed:
removed timestamp from units route
you don't need the timestamp because there isn't a column for it in the database

therealaldo committed:
removed timestamp from inventory route
there isn't a column for timestamp for this table in the database

therealaldo committed:
added .findAll to models along with cleaning the put routes up
using findAll because you want to be able to find all of the items of that kind

Commit:
fixed units model

therealaldo committed:
fixed inventory route
you have to specify you want to pass through the dataValues from the returned db object

therealaldo committed:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

therealaldo committed:
fixed units route
modified the units model and the unit route
you need to pass through the dataValues instead of the entire response object since there is a bunch of extra stuff you don't need

THEJNIGH committed:
More unit testing

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

chrisfortin16 committed:
Restructured readme file

Commit:
modified inventory route to go to specific sku

THEJNIGH committed:
Unit tests working for orderedItems and Order routes

THEJNIGH committed:
Merge branch 'db_create' of github.com:reactivepixel/Gravity into db_create

Commit:
modified units route to go to specific sku

chrisfortin16 committed:
created units.js file for unit testing

Commit:
added inventory unit test

Commit:
Merge branch 'db_create' of https://github.com/reactivepixel/Gravity into db_create

therealaldo committed:
modified inventory unit test

Commit:
fixed syntax errors

THEJNIGH committed:
Fixing tests

therealaldo committed:
working on inventory unit test

chrisfortin16 committed:
changed around __units.js file

chrisfortin16 committed:
One last change from uuid to sku

reactivepixel committed:
Update readme.md
